packages/amplify-util-mock/src/CFNParser/appsync-resource-processor.ts

```diff
@@ -1,235 +1,59 @@
-}
-
-export function graphQLSchemaHandler(resourceName, resource, cfnContext: CloudFormationParseContext, transformResult: any) {
-  return {
-    content: transformResult.schema,
-    path: 'schema.json', // XXX: handle schema folder
-  };
-}
-
-export function graphQLResolverHandler(resourceName, resource, cfnContext: CloudFormationParseContext, transformResult: any) {
-  const { Properties: properties } = resource;
-  const requestMappingTemplate = parseValue(properties.RequestMappingTemplateS3Location, cfnContext).replace(
-    's3://${S3DeploymentBucket}/${S3DeploymentRootKey}/',
-    ''
-  );
-  const responseMappingTemplate = parseValue(properties.ResponseMappingTemplateS3Location, cfnContext).replace(
-    's3://${S3DeploymentBucket}/${S3DeploymentRootKey}/',
-    ''
-  );
-  let dataSourceName;
-  let functions;
-  if (properties.Kind === 'PIPELINE') {
-    functions = (properties.PipelineConfig.Functions || []).map(f => getAppSyncFunctionName(f));
-  } else {
-    dataSourceName = getDataSourceName(properties.DataSourceName, cfnContext.resources);
-  }
-
-  return {
-    dataSourceName,
-    typeName: properties.TypeName,
-    functions,
-    fieldName: properties.FieldName,
-    requestMappingTemplateLocation: requestMappingTemplate,
-    responseMappingTemplateLocation: responseMappingTemplate,
-    kind: properties.Kind || 'UNIT',
-  };
-}
-
-function getDataSourceName(dataSourceName, resources) {
-  // XXX: Util to map data source based on type of intrinsic function
-  let processedDataSourceName;
-  if (typeof dataSourceName === 'string') {
-    return dataSourceName;
-  }
-
-  if (isPlainObject(dataSourceName) && Object.keys(dataSourceName).length === 1) {
-    const intrinsicFn = Object.keys(dataSourceName)[0];
-    if (intrinsicFn === 'Fn::GetAtt') {
-      processedDataSourceName = dataSourceName[intrinsicFn][0];
+export function processApiResources(
+  resources: Record<string, { Type: string; result: any }>,
+  transformResult: any,
+  appSyncConfig: AmplifyAppSyncSimulatorConfig,
+): void {
+  Object.values(resources).forEach(resource => {
+    const { Type: resourceType } = resource;
+    const result: any = resource.result;
+
+    switch (resourceType) {
+      case 'AWS::AppSync::DataSource':
+        appSyncConfig.dataSources.push(result);
+        break;
+      case 'AWS::AppSync::Resolver':
+        appSyncConfig.resolvers.push(result);
+        break;
+      case 'AWS::DynamoDB::Table':
+        appSyncConfig.tables.push(result);
+        break;
+      case 'AWS::AppSync::FunctionConfiguration':
+        appSyncConfig.functions.push(result);
+        break;
+      case 'AWS::AppSync::GraphQLSchema':
+        appSyncConfig.schema = result;
+        break;
+      case 'AWS::AppSync::GraphQLApi':
+        const resource = result as AppSyncAPIProcessedResource;
+        appSyncConfig.appSync.name = resource.name;
+        appSyncConfig.appSync.defaultAuthenticationType = resource.defaultAuthenticationType;
```

Comment:
Isn't it ' authenticationType' as of https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-graphqlapi.html#cfn-appsync-graphqlapi-additionalauthenticationproviders?